processing/src/main/java/org/apache/druid/segment/nested/ScalarDoubleColumnAndIndexSupplier.java

```diff
@@ -258,6 +265,70 @@ public <T> T computeBitmapResult(BitmapResultFactory<T> bitmapResultFactory, boo
     }
   }
 
+  private final class DoubleValueSetIndexes implements ValueSetIndexes
+  {
+    final FixedIndexed<Double> dictionary = doubleDictionarySupplier.get();
+
+    @Nullable
+    @Override
+    public BitmapColumnIndex forSortedValues(@Nonnull List<?> sortedValues, TypeSignature<ValueType> matchValueType)
+    {
+      final boolean matchNull = sortedValues.get(0) == null;
+      final Supplier<ImmutableBitmap> unknownsIndex = () -> {
+        if (!matchNull && dictionary.get(0) == null) {
+          return valueIndexes.get(0);
+        }
+        return null;
+      };
+
+      // values are doubles and ordered in double order
+      if (matchValueType.is(ValueType.DOUBLE)) {
```

Comment:
yea probably, also for the other index implementations.. i can look into it

Reply:
i guess the downside to doing this is that i was hoping at some point to specialize the value dictionaries for primitive values (e.g. #12846) and combining them would make that not very easy since would need a generic parameter instead of java primitive.

Reply:
i have not done this change yet, still thinking about it

Reply:
I am okay with what you decide here. I can see both ways making sense.